extension/chrome/settings/modules/test_passphrase.ts

```diff
@@ -13,46 +12,53 @@ import { Assert } from '../../../js/common/assert.js';
 import { initPassphraseToggle } from '../../../js/common/ui/passphrase_ui.js';
 import { Xss } from '../../../js/common/platform/xss.js';
 import { Url } from '../../../js/common/core/common.js';
+import { View } from '../../../js/common/view.js';
 
 declare const openpgp: typeof OpenPGP;
 
-Catch.try(async () => {
+View.run(class TestPassphrase extends View {
+  private readonly acctEmail: string;
+  private readonly parentTabId: string;
+  private primaryKey: OpenPGP.key.Key | undefined;
 
-  const uncheckedUrlParams = Url.parse(['acctEmail', 'parentTabId']);
-  const acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
-  const parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
-
-  await initPassphraseToggle(['password']);
-
-  const [primaryKi] = await Store.keysGet(acctEmail, ['primary']);
-  Assert.abortAndRenderErrorIfKeyinfoEmpty(primaryKi);
-
-  const { keys: [key] } = await openpgp.key.readArmored(primaryKi.private);
-  if (!key.isFullyEncrypted()) {
-    const setUpPpUrl = Url.create('change_passphrase.htm', { acctEmail, parentTabId });
-    Xss.sanitizeRender('#content', `<div class="line">No pass phrase set up yet: <a href="${setUpPpUrl}">set up pass phrase</a></div>`);
-    return;
+  constructor() {
+    super();
+    const uncheckedUrlParams = Url.parse(['acctEmail', 'parentTabId']);
+    this.acctEmail = Assert.urlParamRequire.string(uncheckedUrlParams, 'acctEmail');
+    this.parentTabId = Assert.urlParamRequire.string(uncheckedUrlParams, 'parentTabId');
   }
 
-  $('#password').keydown(event => {
-    if (event.which === 13) {
-      $('.action_verify').click();
+  async render() {
+    const [keyInfo] = await Store.keysGet(this.acctEmail, ['primary']);
+    Assert.abortAndRenderErrorIfKeyinfoEmpty(keyInfo);
+    await initPassphraseToggle(['password']);
+    this.primaryKey = (await openpgp.key.readArmored(keyInfo.private)).keys[0];
```

Comment:
Can write as `this.primaryKey = await Pgp.key.read(keyInfo.private);`

Reply:
That would be much better.
Thanks, Will do!